torchgeo/models/seq2seq.py

```diff
@@ -0,0 +1,192 @@
+        return hidden, cell
+
+
+class LSTMDecoder(nn.Module):
+    """Decoder for LSTM Seq2Seq."""
+
+    def __init__(
+        self,
+        input_size: int,
+        hidden_size: int,
+        output_size: int,
+        target_indices: list[int] | None = None,
+        num_layers: int = 1,
+        output_sequence_len: int = 1,
+        teacher_force_prob: float | None = None,
+    ) -> None:
+        """Initialize a new LSTMDecoder.
+
+        Args:
+            input_size: The number of features in the input.
+            hidden_size: The number of features in the hidden state.
+            output_size: The number of features output by the decoder.
+            target_indices: Indices of the target features in the dataset.
+                If None, uses all features passed to the decoder. Defaults to None.
+            num_layers: Number of LSTM layers. Defaults to 1.
+            output_sequence_len: The number of steps to predict forward. Defaults to 1.
+            teacher_force_prob: Probability of using teacher forcing. If None, does not
+                use teacher forcing. Defaults to None.
+        """
+        super().__init__()
+        self.lstm = nn.LSTM(input_size, hidden_size, num_layers, batch_first=True)
+        self.fc = nn.Linear(hidden_size, output_size)
+        self.output_size = output_size
+        self.target_indices = target_indices
+        self.output_sequence_len = output_sequence_len
+        self.teacher_force_prob = teacher_force_prob
+
+    def forward(self, inputs: Tensor, hidden: Tensor, cell: Tensor) -> Tensor:
+        """Forward pass of the decoder.
+
+        Args:
+            inputs: Input sequence of shape (b, sequence length, input_size).
+            hidden: hidden state from the encoder.
+            cell: cell state from the encoder.
+
+        Returns:
+            Output sequence of shape (b, output_sequence_len, output_size).
+        """
+        batch_size = inputs.shape[0]
+        outputs = torch.zeros(batch_size, self.output_sequence_len, self.output_size)
+
+        current_input = inputs[:, 0:1, :]
+
+        for t in range(self.output_sequence_len):
+            _, (hidden, cell) = self.lstm(current_input, (hidden, cell))
+            last_layer_hidden = hidden[-1:]
+            output = self.fc(last_layer_hidden)
+            output = output.permute(1, 0, 2)  # put batch dimension first
+            outputs[:, t : t + 1, :] = output
+            current_input = inputs[:, t : t + 1, :].clone()
+            teacher_force = (
+                random.random() < self.teacher_force_prob
+                if self.teacher_force_prob is not None
+                else False
+            )
+            if not teacher_force:
+                if self.target_indices:
+                    current_input[:, :, self.target_indices] = output
+                else:
+                    current_input = output
+
+        return outputs
+
+
+class LSTMSeq2Seq(nn.Module):
```

Comment:
Let's rename this to `Seq2Seq` and add a new parameter:
```python
cell: Literal['rnn', 'gru', 'lstm'] = 'lstm'
```
That way we can easily support all 3:
```python
match cell:
    case 'rnn':
        self.rnn = nn.RNN(...)
    case 'gru':
        self.rnn = nn.GRU(...)
    case 'lstm':
        self.rnn = nn.LSTM(...)
```